crates/hera-exex/src/exex.rs

```diff
@@ -0,0 +1,60 @@
+//! Module for the Hera Execution Extension.
+
+use std::sync::Arc;
+
+use eyre::{bail, Result};
+use reth_exex::{ExExContext, ExExEvent};
+use reth_node_api::FullNodeComponents;
+use superchain_registry::RollupConfig;
+use tracing::{debug, info};
+
+use crate::cli::HeraArgsExt;
+
+/// The Hera Execution Extension.
+#[derive(Debug)]
+pub struct HeraExEx<Node: FullNodeComponents> {
+    /// The rollup configuration
+    cfg: Arc<RollupConfig>,
+    /// The context of the Execution Extension
+    ctx: ExExContext<Node>,
+}
+
+#[allow(unused)]
+impl<Node: FullNodeComponents> HeraExEx<Node> {
+    /// Creates a new instance of the Hera Execution Extension.
+    pub async fn new(ctx: ExExContext<Node>, args: HeraArgsExt, cfg: Arc<RollupConfig>) -> Self {
+        Self { ctx, cfg }
+    }
+
+    /// Wait for the L2 genesis L1 block (aka "origin block") to be available in the L1 chain.
+    async fn wait_for_l2_genesis_l1_block(&mut self) -> Result<()> {
+        loop {
+            if let Some(notification) = self.ctx.notifications.recv().await {
```

Comment:
Then these `ctx` references can use the `DriverContext` trait which would provide a way to get notifications.